blobfuse/directoryapis.cpp

```diff
@@ -102,7 +100,7 @@ int azs_readdir(const char *path, void *buf, fuse_fill_dir_t filler, off_t, stru
     }
 
     errno = 0;
-    std::vector<std::pair<std::vector<list_blobs_hierarchical_item>, bool>> listResults = list_all_blobs_hierarchical(str_options.containerName, "/", pathStr.substr(1));
+    std::vector<std::pair<std::vector<list_segmented_item>, bool>> listResults = storage_client->ListAllItemsSegmented(pathStr.substr(1), "/");
```

Comment:
For this I need more input on how it will break. Now that we have two different ways to fetch data (BlockBlob and ADLS) storage_client is just abstraction representing those. Actual work is happening in BlockBlobBfsClient::ListAllItemsSegmented method. Can you kindly check there and suggest if we are missing something on continuation token front.